packages/extension-huggingface/test/textGeneration.spec.ts

```diff
@@ -0,0 +1,191 @@
+    async () => {
+      const token = process.env['HF_ACCESS_TOKEN'];
+      const { compileAndLoad, execute } = await getTestCompiler({
+        extensions: { huggingface: path.join(__dirname, '..', 'src') },
+        huggingface: {
+          accessToken: token,
+        },
+      });
+
+      const sql = `{% set data = ${JSON.stringify(
+        repositories
+      )} %}SELECT {{ data | huggingface_text_generation("Not contains query argument!") }}`;
+
+      // Act
+      await compileAndLoad(sql);
+
+      // Assert
+      await expect(execute({})).rejects.toThrow(
+        'Must provide "query" keyword argument'
+      );
+    },
+    50 * 1000
+  );
+
+  it('Should throw error when input value not be array of object', async () => {
+    const token = process.env['HF_ACCESS_TOKEN'];
+    const { compileAndLoad, execute } = await getTestCompiler({
+      extensions: { huggingface: path.join(__dirname, '..', 'src') },
+      huggingface: {
+        accessToken: token,
+      },
+    });
+
+    const sql = `{% set data = 'not-array-data' %}SELECT {{ data | huggingface_text_generation(query="Does the filter work or not") }}`;
+
+    // Act
+    await compileAndLoad(sql);
+
+    // Assert
+    await expect(execute({})).rejects.toThrow(
+      'Input value must be an array of object'
+    );
+  });
+
+  it(
+    'Should throw error when not provide access token',
+    async () => {
+      const { compileAndLoad, execute } = await getTestCompiler({
+        extensions: { huggingface: path.join(__dirname, '..', 'src') },
+        huggingface: {
+          accessToken: '',
+        },
+      });
+
+      const sql = `{% set data = ${JSON.stringify(
+        repositories
+      )} %}SELECT {{ data | huggingface_text_generation("${faker.internet.password()}") }}`;
+
+      // Act
+      await compileAndLoad(sql);
+
+      // Assert
+      await expect(execute({})).rejects.toThrow('please given access token');
+    },
+    50 * 1000
+  );
+  
+  it(
+    'Should not throw when pass the "query" argument by dynamic parameter through HuggingFace recommended model',
+    async () => {
+      const token = process.env['HF_ACCESS_TOKEN'];
+      const { compileAndLoad, execute } =
+        await getTestCompiler({
+          extensions: { huggingface: path.join(__dirname, '..', 'src') },
+          huggingface: {
+            accessToken: token,
+          },
+        });
+
+      const sql = `{% set data = ${JSON.stringify(
+        repositories
+      )} %}SELECT {{ data | huggingface_text_generation(query=context.params.value, wait_for_model=true, use_cache=false) }}`;
+
+      
+      await compileAndLoad(sql);
+      // Assert
+      await expect(execute({ value: 'what repository has most stars?' })).resolves.not.toThrow();
+    },
+    100 * 1000
+  );
+
+  // Skip the test case because the "meta-llama/Llama-2-13b-chat-hf" model need to upgrade your huggingface account to Pro Account by paying $9 per month
```

Comment:
Is there any model that is free and can be used for testing?
If the structure of API response payload is  the same, I think it could be used for testing.